### PR TITLE
Fix unuran builds for older compilers

### PR DESCRIPTION
**Pull request by isuruf**
#### Additional information

cc @tirthasheshpatel, @chrisb83, @rgommers 

Otherwise builds fail with undefiend symbol `isfinite`.
See https://dev.azure.com/numpy/numpy/_build/results?buildId=19164&view=results

### CODE REVIEW

**Review by tirthasheshpatel — COMMENTED** [review]
Hi, @isuruf. Thanks for fixing this!

Did you try to build with `("HAVE_DECL_ISFINITE", "0")` on line 75? UNU.RAN is completely ANSI C compatible. So, we should not have a problem by defining all the C99 functions as unavailable. Even better, we can add a check for them in the pre-build hook (it's OK if you don't do that here).

Sorry, I forgot to check for some C99/C95 functions while building. Let me know if you find more of these. I will review this more thoroughly later today.

**Comment by isuruf** [comment]
> Did you try to build with ("HAVE_DECL_ISFINITE", "0") on line 75?

Nope. I don't have a scipy build locally and just trying at https://github.com/MacPython/scipy-wheels. As you can see, the CI there is failing.

**Comment by tirthasheshpatel** [comment]
I have proposed #14632 to see if SciPy builds with definitions for C99 functions removed. @isuruf If you can locally replicate the failures on MacPython/scipy-wheels, try to build it with #14632.

@mckib2 would you be able to help confirm this and gh-14632?

**Comment by isuruf** [comment]
@tirthasheshpatel, I redid the PR

File: scipy/stats/_unuran/setup.py [hunk 1]
```diff
@@ -105,7 +105,8 @@ def configuration(parent_package="", top_path=None):
         ("STDC_HEADERS", "1"),
         ("UNUR_ENABLE_INFO", "1"),
         ("VERSION", '"%s"' % UNURAN_VERSION),
-        ("HAVE_CONFIG_H", "1")
+        ("HAVE_CONFIG_H", "1"),
+        ("_ISOC99_SOURCE", "1"),
```
Comment (tirthasheshpatel) [inline scipy/stats/_unuran/setup.py:109]: How is this different than adding a `-std=c99` flag? Is it possible that the compiler doesn't recognize the `-std=c99` flag but understands `_ISOC99_SOURCE`?

Comment (tirthasheshpatel) [reply scipy/stats/_unuran/setup.py:109]: Never mind, your commit message answers the question:

> For gcc 4.8.2 with -std=c99 _ISOC99_SOURCE is not set

Comment (tirthasheshpatel) [reply scipy/stats/_unuran/setup.py:109]: One last comment: Defining `("_ISO99_SOURCE", "1")` would read `#define _ISO99_SOURCE 1`. I think we should rather do this:

```suggestion
        ("_ISOC99_SOURCE", None),
```

which is equivalent to `#define _ISOC99_SOURCE`.

Comment (isuruf) [reply scipy/stats/_unuran/setup.py:109]: It's usually set to `1`. For eg: in `/usr/include/features.h` it is defined as `# define _ISOC99_SOURCE 1`

Comment (tirthasheshpatel) [reply scipy/stats/_unuran/setup.py:109]: Oh, OK. Thanks for clarifying!

**Review by tirthasheshpatel — APPROVED** [review]
LGTM now! This builds on SciPy and fixes the failures on MacPython/scipy-wheels. @mckib2 @rgommers Feel free to add your review and merge.

**Review by rgommers — APPROVED** [review]
Great, thanks @isuruf and @tirthasheshpatel!

**Pull request merged**